### PR TITLE
Fall back to co-location heuristic if sourcemap url appears remote

### LINKED ISSUE #1870: `sourcemaps inject` should fall back to co-located source maps if exact path can't be found
### Environment

What version of the CLI and what OS are you running?

Sentry CLI 1.23.0, macOS Ventura 13.4

### Steps to Reproduce

Similar to #1846:
1. Create a `.js` file with a corresponding `.js.map` file in the same directory
2. Add a source map comment pointing to a path that doesn't exist locally; e.g.
   ```
   //# sourceMappingURL=/_static/build/foo.js.map
   ```
   Note: this is a mostly-real example from a starter Remix app. The path here is the default when creating a new app from the [grunge stack](https://github.com/remix-run/grunge-stack), but this issue will occur for any static assets path, whether or not it has a URL scheme and domain.
3. Run `sentry-cli sourcemaps inject foo.js*`

Alternatively, though it was created to repro a different bug, you can also repro this bug by following the steps [here](https://github.com/brettdh/sentry-remix-lambda-esm-variables), ignoring the AWS bits and running `npm run build` instead of any of the other scripts. The `sentry-cli sourcemaps inject` step will show missing debug IDs for the sourcemap files.

### Expected Result

JS file and sourcemap file both have debug ID injected

### Actual Result

Only JS file has debug ID injected

### Logs

When filing a bug report, please attach debug logs, either by using `--log-level=debug` flag or `SENTRY_LOG_LEVEL=debug` environment variable.

### PR DESCRIPTION
**Pull request by brettdh**
Applies a heuristic to guess that a sourcemap URL is a remote URL path (without scheme and domain):
1. The string starts with `/`
2. The string is not a path to an existing local file

Fixes #1870.

### CODE REVIEW

File: src/utils/sourcemaps.rs [hunk 1]
```diff
@@ -247,6 +247,19 @@ fn is_remote_url(url: &str) -> bool {
     };
 }
 
+/// Return true iff url appears to be a URL path.
+/// Most often, a URL path will begin with `/`,
+/// particularly in the case of static asset collection and hosting,
+/// but such a path is very unlikely to exist in the local filesystem.
+fn is_url_path(url: &str) -> bool {
+    url.starts_with('/') && !Path::new(url).exists()
```
Comment (anonrig) [inline src/utils/sourcemaps.rs:255]: As far as I know, `path.exists()` makes a system call. Can we avoid having validation only filesystem calls, and also achieve the same result of `is_url_path` purpose? `Path::new(url).is_absolute()` for example?

File: src/utils/sourcemaps.rs [hunk 2]
```diff
@@ -247,6 +247,19 @@ fn is_remote_url(url: &str) -> bool {
     };
 }
 
+/// Return true iff url appears to be a URL path.
```
Comment (anonrig) [inline src/utils/sourcemaps.rs:250]:
```suggestion
/// Return true if url appears to be a URL path.
```

**Comment by brettdh** [comment]
This is still on my todo list.

**Comment by szokeasaurusrex** [comment]
> ...but the stalebot did not remove the label. 🤔

I removed "Status: Stale" and added "Status: In Progress." Hopefully StaleBot leaves the PR alone going forward

**Comment by szokeasaurusrex** [comment]
@brettdh is this still in progress?

**Comment by brettdh** [comment]
Hmm, I thought I'd responded to these changes already, sorry. I'll try to push an update today.

File: src/utils/sourcemaps.rs (hunk 1 above)
Comment (brettdh) [reply src/utils/sourcemaps.rs:255]: The file non-existence check is an important part of `is_url_path`, as it prevents treating a local path as a remote URL path (without the `https://` part).

**Comment by brettdh** [comment]
Responded to the comments above and resolved one. Please let me know if more is required.

File: .vscode/settings.json [hunk 3]
```diff
@@ -8,6 +8,6 @@
     "eslint.autoFixOnSave": true,
     "editor.tabSize": 2,
     "editor.codeActionsOnSave": {
-        "source.fixAll.eslint": true
+        "source.fixAll.eslint": "explicit"
```
Comment (szokeasaurusrex) [inline .vscode/settings.json:11]: Guessing this file was accidentally committed

```suggestion
        "source.fixAll.eslint": true
```

**Review by szokeasaurusrex — APPROVED** [review]
Looks good, thanks for the contribution!

File: .vscode/settings.json (hunk 3 above)
Comment (brettdh) [reply .vscode/settings.json]: Whoops! Yep, unintentional. Thanks!

**Pull request merged**